### PR TITLE
Add filters to git push event to trigger email workflow when a PR is closed and merged.

### PR DESCRIPTION
**Pull request by bhavanijayakumaran**
This is a follow up for:
 #20794
 
 Email step is not triggered after the fix #20818.
 There could be a couple of things that might have caused the email step to be skipped:
The on branch filter(that filters on branch main) or the if PR is merged check.

 Trying with the if condition commented out to see it the workflow step is triggered for all push events to main branch.

### CODE REVIEW

File: .github/workflows/email.yml [hunk 1]
```diff
@@ -3,21 +3,25 @@ name: email
 on:
   push:
     branches:
-      - main 
+     # - main
```
Comment (bhavanijayakumaran) [inline .github/workflows/email.yml:6]: Commented out to see if this works when we merge a PR to main from forked repository.
If this patch works will remove this commented line.

File: .github/workflows/email.yml [hunk 2]
```diff
@@ -3,21 +3,25 @@ name: email
...
-      - main 
+     # - main 
       
 jobs:
   
   
   send_email:
    # run this workflow step only when the PR is merged 
-    if:  github.event.pull_request.merged == true 
+    #if:  github.event.pull_request.merged == false
```
Comment (bhavanijayakumaran) [inline .github/workflows/email.yml:13]: Commented out to see if this is what is causing the email step to be skipped. 
Will remove this line if this dix works in a followup review.

**Comment by bhavanijayakumaran** [comment]
Tried closing a PR with close filter and it works as expected.

https://github.com/bhavanijayakumaran/chapel/actions/runs/3238100043

**Review by riftEmber — CHANGES_REQUESTED** [review]
As far as I can tell this will trigger on a PR that is closed even if it is not merged, which I don't think is what we want. I think the `if_merged` step you wrote has the right idea and that kind of conditional could be used to avoid sending the email if it is not a merge.

Additionally, it looks like Elliot's suggestion in https://github.com/chapel-lang/chapel/pull/20794#discussion_r991582020 to remove some extra newlines didn't make it in -- I would like to echo that suggestion as I also think it would be good for readability to cut down on the extra line breaks.

**Comment by bhavanijayakumaran** [comment]
@riftEmber I have checks to check for both closed and merged. 

name: email
on:
    pull_request:
      types:
        **- closed**   
jobs:
  
  send_email:
   # run this workflow step only when the PR is merged 
    if:  **github.event.pull_request.merged == true** 
    runs-on: ubuntu-latest
    
    I will remove the extra new lines.

**Review by riftEmber — APPROVED** [review]
Sorry, missed the condition on the job as a whole. Looks good!

**Comment by bhavanijayakumaran** [comment]
Np @riftEmber. Thanks for the review!

**Comment by bhavanijayakumaran** [comment]
Ok, the email workflow is triggered but the actions secrets are not available for git workflow.

I will try to fix it, if it takes a while I will just revert the PR.
https://github.com/chapel-lang/chapel/actions/runs/3244107990/jobs/5319807737

**Pull request merged**